src/http-api/fetch.ts

```diff
@@ -293,6 +301,22 @@ export class FetchHttpApi<O extends IHttpOpts> {
         return res as ResponseType<T, O>;
     }
 
+    private clearUrlParamsForLogs(url: URL | string): string {
+        try {
+            let asUrl: URL;
+            if (typeof url === "string") {
+                asUrl = new URL(url);
+            } else {
+                asUrl = url;
+            }
+            // get just the path to remove any potential url param that could have
+            // some potential secrets
+            return asUrl.pathname;
```

Comment:
could we include the hostname here, just to avoid confusion where we think the request is going to the HS but it turns out to be going somewhere else?

```suggestion
            return asUrl.origin + asUrl.pathname;
```